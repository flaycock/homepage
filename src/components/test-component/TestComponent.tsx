import type { FC } from "react";
import { Heading, Text } from "@chakra-ui/react";
import styled from "styled-components";

import type { TestInput } from "./TestComponent.types";

const TestComponentWrapper = styled.div`
  margin: auto;
  padding: 16px;
  text-align: center;
`;

export const TestComponent: FC<TestInput> = ({ text }) => (
	<TestComponentWrapper data-test-id="test-component">
		<Heading>This website is currently being rebuilt</Heading>
		{text && <Text data-test-id="test-component-text">{text}</Text>}
	</TestComponentWrapper>
);
