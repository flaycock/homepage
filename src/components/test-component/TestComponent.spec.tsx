import { render as rtlRender, screen } from "@testing-library/react";

import { Provider } from "../ui/provider";
import { TestComponent } from "./TestComponent";

const render = (text?: string) => rtlRender(
  <Provider>
    <TestComponent text={text} />
  </Provider>    
);

describe("TestComponent", () => {
  it("renders", () => {
    render("test");

    expect(screen.getByTestId("test-component")).toMatchSnapshot();
  });

  it("does not render a Text component if no text is provided", () => {
    render();

    expect(screen.queryByTestId("test-component-test")).not.toBeInTheDocument();
  });
});
