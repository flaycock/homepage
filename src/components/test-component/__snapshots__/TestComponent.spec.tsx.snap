// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent renders 1`] = `
<div
  class="sc-bRKDuR hXXjjt"
  data-test-id="test-component"
>
  <h2
    class="chakra-heading css-eprwor"
  >
    This website is currently being rebuilt
  </h2>
  <p
    class="css-0"
    data-test-id="test-component-text"
  >
    test
  </p>
</div>
`;
